ext/ffi/00_ffi.js

```diff
@@ -213,6 +247,59 @@
     return type === "i64" || type === "isize";
   }
 
+  function isStruct(type) {
+    return typeof type === "object" && type !== null && ("struct" in type);
+  }
+
+  function getTypeSize(type, cache = new WeakMap()) {
+    if (isStruct(type)) {
+      const cached = cache.get(type);
+      if (cached) {
+        if (cached === null) {
```

Comment:
This isn't possible, maybe one was supposed to be `cache`?
Also, `cache.get(...)` would return `undefined` if not present, as opposed to `null`.

Reply:
While calculating struct size I'm setting its size to null explicitly. So for example if we're calculating size of a struct Rect which has a field Rect (itself) in it. When calculating outer struct's size if we try to look up the field (which is the outer struct itself) in cache again it'll return null which means it's recursive. Basically this is cache + recursive check combined.

Reply:
Oops, got it.
Ignore this; Thanks for the explanation.